Godot XR Update Sep 2023 (#693)
godotengine/godot-website: collections/_article/godot-xr-update-aug-2023.md

---
title: "Godot XR update - August 2023"
excerpt: "Updates on various things XR in Godot, a new version of the OpenXR plugin, a new version of the tools library and an update on Godot 4 support."
categories: ["progress-report"]
author: Bastiaan Olij
image: /storage/blog/covers/godot-xr-update-aug-2023.webp
date: 2023-08-27 11:00:00
---

With the last grant provided by Meta Reality Labs having ended in December 2022 the core team's focus on XR has been limited.

There is still plenty happening on the XR front to warrant another update.

We again thank Meta for the support past and present that they have been able to provide.

## New plugin structure on Android

> Thanks to the work of Fredia Huya-Kouadio Godots Android support for the upcoming Godot 4.2 release has had a big refactor of its plugin structure. This means that the Godot OpenXR loaders plugin has been streamlined and is now able to support newer OpenXR extensions that require platform specific feature tags and permissions.

Reviewer: ```suggestion
Thanks to the work of Fredia Huya-Kouadio, Godot's Android support for the upcoming Godot 4.2 release has had a big refactor of its plugin structure. This means that the Godot OpenXR loaders plugin has been streamlined and is now able to support newer OpenXR extensions that require platform specific feature tags and permissions.
```

A separate post will provide more details about the new Android plugin structure, and how Godot developers can leverage it.

## GDExtension support for OpenXR Extensions

OpenXR regularly sees new vendor extensions being added for functionality that is only available on a single platform or where there are other reasons that prevent adding support to the core of Godot.

> Migeran have upstreamed their improvements to Godot to allow these extensions to be implemented through Godots GDExtension system.

Reviewer: ```suggestion
Migeran have upstreamed their improvements to Godot to allow these extensions to be implemented through Godot's GDExtension system.
```

> Currently they are working on a plugin that implements Metas set of scene and spatial extensions.

Reviewer: ```suggestion
Currently they are working on a plugin that implements Meta's set of scene and spatial extensions.
```

> ## OpenXR MacOS Support

Reviewer: ```suggestion
## OpenXR macOS Support
```

> OpenXR added MacOS support to the latest specification. While XR runtimes are still under development Godot now supports this functionality.

Reviewer: ```suggestion
OpenXR added macOS support to the latest specification. While XR runtimes are still under development Godot now supports this functionality.
```

This also means that the editor functionality, such as the action map editor, is now available on Mac which makes it possible to build Godot XR applications for Android devices using Macs.

## Godot involvement in the OpenXR working group

Thanks to Valve’s continued sponsorship, Bastiaan Olij’s membership in the Khronos OpenXR working group has been extended for another year.

Having game engine representation in the working group is of utmost importance in ensuring the standard caters to the needs of XR game developers.

> ## Godot TiltFive Support

Reviewer: ```suggestion
## Godot Tilt Five Support
```

Contributor Patrick Down has been hard at work making the Tilt Five work with Godot. The Tilt Five system now works fully with Godot 4.1 stable however still requires Tilt Five drivers that are in beta. An official release of the Tilt Five plugin is expected soon.

You can download the beta of the plugin from [the TiltFiveGodot4 repository](https://github.com/GodotVR/TiltFiveGodot4/).

## Meta provides Quest Pros to the team

Meta is providing Quest Pros to contributors Bastiaan Olij, David Snopek and Malcolm Nixon. This will help in developing various newer features available on these devices. 

## Status update Godot XR Tools

> Our XR Tools library has seen a steady flow of improvements over the past months and has some really interesting improvements in the pipeline. We release version 4.2.0 on the 22nd of August with a long list of improvements.

Reviewer: ```suggestion
Our XR Tools library has seen a steady flow of improvements over the past months and has some really interesting improvements in the pipeline. We released version 4.2.0 on the 22nd of August with a long list of improvements.
```

Development has now moved to Godot 4 with the Godot 3 version of the library only receiving occasional fixes.

Thanks to contributor DigitalN8m4r3 the demo has undergone a nice visual overhaul.
<add image>

> He has also contributed a new audio system to XR Tools that makes it easy to create audio que’s as a player moves through the virtual space and interacts with various objects.

Reviewer: ```suggestion
He has also contributed a new audio system to XR Tools that makes it easy to create audio cues as a player moves through the virtual space and interacts with various objects.
```

---
title: "Godot XR update - August 2023"
excerpt: "Updates on various things XR in Godot, a new version of the OpenXR plugin, a new version of the tools library and an update on Godot 4 support."
categories: ["progress-report"]
author: Bastiaan Olij
image: /storage/blog/covers/godot-xr-update-aug-2023.webp
date: 2023-08-27 11:00:00
---

> With the last grant provided by Meta Reality Labs having ended in December 2022 the core team's focus on XR has been limited.
> 
> There is still plenty happening on the XR front to warrant another update.
> 
> We again thank Meta for the support past and present that they have been able to provide.

Reviewer 2: I would shift the focus away from Meta in these first couple of paragraphs. There is no reason to thank them for a grant that didn't fund the described work. Similarly, we want to highlight both that we need an XR grant to shift the focus back to XR, and that the grant can come from anyone (It would be great if it came from Meta as we have worked well with them in the past, but other companies are free to step up as well).

I would go with something a little bit subtler
```suggestion
XR work has slowed down this year due to the XR-specific line of funding running out. However, there is still plenty happening on the XR front to warrant another update.
```

Author: I didn't want to come across too harsh, but yeah us currently not having any funding for XR has had a measurable impact which is why I wanted to highlight that. I like your rewording

Having game engine representation in the working group is of utmost importance in ensuring the standard caters to the needs of XR game developers.

## Godot Tilt Five Support

Contributor Patrick Down has been hard at work making the Tilt Five work with Godot. Tilt Five also provided Bastiaan Olij with a headset to help finalise the work being done. The Tilt Five system now works fully with Godot 4.1 stable however still requires Tilt Five drivers that are in beta. An official release of the Tilt Five plugin is expected soon.

You can download the beta of the plugin from [the TiltFiveGodot4 repository](https://github.com/GodotVR/TiltFiveGodot4/).

## Meta provides Quest Pros to the team

Meta is providing Quest Pros to contributors Bastiaan Olij, David Snopek and Malcolm Nixon. This will help in developing various newer features available on these devices.

## Godot PICO Support

PICO has been working closely with the Godot XR Team to ensure Godot runs properly on PICO devices as well. This included making a PICO 4 headset available to the team.

> With the latest OS update on the PICO headset both compatibility and Vulkan mobile renderers are working well on the device.

Reviewer: ```suggestion
With the latest OS update on the PICO headset both Compatibility and Vulkan mobile renderers are working well on the device.
```
So it's clearer that compatibility is a renderer.
